src/ConnectionFX3/ConnectionFX3.cpp

```diff
@@ -509,17 +509,10 @@ bool ConnectionFX3::WaitForReading(int contextHandle, unsigned int timeout_ms)
     status = contexts[contextHandle].EndPt->WaitForXfer(contexts[contextHandle].inOvLap, timeout_ms);
 	return status;
     #else
-    auto t1 = chrono::high_resolution_clock::now();
-    auto t2 = t1;
-
+    //blocking not to waste CPU
     std::unique_lock<std::mutex> lck(contexts[contextHandle].transferLock);
-    while(contexts[contextHandle].done.load() == false && std::chrono::duration_cast<std::chrono::milliseconds>(t2 - t1).count() < timeout_ms)
-    {
-        //blocking not to waste CPU
-        contexts[contextHandle].cv.wait_for(lck, chrono::milliseconds(timeout_ms));
-        t2 = chrono::high_resolution_clock::now();
-    }
-    return contexts[contextHandle].done.load() == true;
+    contexts[contextHandle].cv.wait_for(lck, chrono::milliseconds(timeout_ms), [&](){return contexts[contextHandle].done.load();});
+    return contexts[contextHandle].done == true;
```

Comment:
On a timeout, while unlikely, contexts[contextHandle].done value could have changed, since you are not loading it anymore.